compiler/src/dotty/tools/dotc/core/TypeComparer.scala

```diff
@@ -401,7 +401,13 @@ class TypeComparer(@constructorOnly initctx: Context) extends ConstraintHandling
         compareErasedValueType
       case ConstantType(v2) =>
         tp1 match {
-          case ConstantType(v1) => v1.value == v2.value && recur(v1.tpe, v2.tpe)
```

Comment:
MEMO 

I tried to modify the Constants.scala below to change the processing of the `==` method, but it didn't work.

https://github.com/scala/scala3/blob/a5a7644dd900ee7af8c36e87205b4d0b64275878/compiler/src/dotty/tools/dotc/core/Constants.scala#L56

Reply:
The line should use equals, which is already correct, see equalHashValue comment.
```
case ConstantType(v1) => v1.value.equals(v2.value) && recur(v1.tpe, v2.tpe)
```